### PR TITLE
Document spec-file-from-Git-repo feature (SOFTWARE-2962)

### PR DESCRIPTION
**Pull request by matyasselmeci**

### CODE REVIEW

**Comment by matyasselmeci** [comment]
As a side note, this doc needs a lot reorganization: it has a very deep tree structure, which is bad because by the time you reach h6's, the header font is smaller than the body text.

File: docs/software/rpm-development-guide.md [hunk 1]
```diff
@@ -143,16 +148,58 @@ where:
...
+|:---------|:---------------------------------|:---------------------------------------------------|
+| `<URL>`  | Location of the Git repo         | `https://github.com/opensciencegrid/osg-build.git` |
+| `<NAME>` | Name of the software (optional)  | `osg-build`                                        |
+| `<TAG>`  | Git tag to use                   | `v1.11.2`                                          |
+| `<HASH>` | Full 40-char Git hash of the tag | `5bcf48c442d21b1e8c93a468d884f84122f7cc9e`         |
+
+!!! note
+    `<NAME>` is optional; if not present, OSG-Build will use the last component of the URL, without the `.git` suffix.
+
+    The tarball will be called `<NAME>-<VERSION>.tar.gz` where `<VERSION>` is `<TAG>` without the `v` prefix.
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]: Does this still work if the tag isn't prefixed by `v`?

File: docs/software/rpm-development-guide.md [hunk 2]
```diff
@@ -143,16 +148,58 @@ where:
...
 
-For example, the reference file for `globus-common`'s source tarball is named `epel.srpm.source` and might contain:
+OSG-Build 1.11.2 or later is required to use this feature.
 
-    globus-common/16.4/globus-common-16.4-1.el6.src.rpm
-    # Downloaded from 'http://dl.fedoraproject.org/pub/epel/6/SRPMS/globus-common-16.4-1.el6.src.rpm'
+!!! warning
+    OSG software policy requires that all Git repos used for building software have mirrors at the UW.
+    Many software repos under the [opensciencegrid GitHub organization](https://github.com/opensciencegrid) are already mirrored.
+    If you are uncertain, or have a new project that you want mirrored, send email to <osg-software@opensciencegrid.org>.
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]: Could we put this warning at the top of the git repos section?

**Comment by matyasselmeci** [comment]
Fixed. Can you take another look?

File: docs/software/rpm-development-guide.md (hunk 2 above)
Comment (matyasselmeci) [reply docs/software/rpm-development-guide.md]: Done.

**Pull request merged**